docs/docs/adrs/adr-018-permissionless-ics.md

```diff
@@ -0,0 +1,342 @@
+the chain can launch and moves to the _launched_ phase. While in launched phase, a Top N consumer chain can choose to modify
+its parameters through a `MsgConsumerModification` and an Opt In chain can change its parameters by issuing the `MsgUpdateConsumerChain` transaction.
+
+Lastly, a Top N chain can choose to exit ICS by issuing a `MsgConsumerRemoval` and an Opt In chain can issue a transaction to stop the chain.
+After some period of time (e.g., provider's unbonding period), all state related to the stopped consumer chain can be removed. We
+keep track of the consumer chain's state for some period, so that we are able to punish validators for misbehaviours that occurred before the consumer chain stopped.
+
+Note that everything described so far and everything that follows applies to consumer chains that transition from standalone chains as well.
+
+### From `chainId` to `consumerId`
+A hindrance in moving to Permissionless ICS is [chain-id squatting](https://forum.cosmos.network/t/pss-permissionless-vs-premissioned-lite-opt-in-consumer-chains/12984/17).
+In a permissionless setting, anyone could issue a transaction to launch a consumer chain with a `chainId` that might already be used by some other consumer chain. This is a problem
+because in the current design the majority of stored state for a consumer chain is indexed using the `chainId` as the key (e.g.,
+see [key used to store client ids](https://github.com/cosmos/interchain-security/blob/v5.1.0/x/ccv/provider/types/keys.go#L245)).
+To tackle this problem, in Permissionless ICS, we introduce the `consumerId` that defines a consumer chain and is simply
+an increasing counter (i.e., `counter`), thus we can support multiple consumer chains with the same `chainId`.
+Another way to understand this is with an analogy between consumer chains and IBC clients: Imagine having multiple IBC clients
+that each point to different consumer chains, but all share the exact same `chainId`. It is then up to the user to select the
+appropriate client (i.e., `clientID`) based on the actual chain they want to communicate with. Similarly, there can be multiple
+consumer chains with the exact same `chainId`, and it is the responsibility of the validator to choose the one they wish
+to interact with by providing the right `consumerId`.
+
+Although a `counter`, we store it as a `string` and not as an integer (e.g., `uint64`) for three reasons:
+i) avoid extensive refactorings when moving from `chainId` that is of type `string` to `consumerId`,
+ii) simplify migrations because we do not need to keep different methods at the same time (e.g., `ChainToClientKey(string)`
+and `ChainToClientKeyV2(uint64)` to do something like `client := Get(ChainToClientKey("currentConsumerChain))` and
+`Set(ChainToClientKeyV2(1234), client)`, and iii) allow for easy ways to incorporate more data to a `consumerId` if needed in the future.
```

Comment:
Not sure that this is needed. Also, I'm not sure I agree with this. Why does it have to be a string. What's the problem with having `ChainToClientKey(uint64)`? Why would you need the string version? In the migration code you can just use the definition of the old `ChainToClientKey(string)` (we do this in a lot of migrations).